### ISSUE TITLE
Support transformers 5.x

### ISSUE THREAD

**Issue by piyifan123** [ByteDance-Seed/VeOmni#468]
### Feature request

1. Add a way to run uv with transformers 5.x as an optional path.
2. Separate model loading based on transformers version in the env.
3. Support code gen based modeling (more details to come).

Known problems with transformers v5 (keep updating):
1. tokenizer refactored significantly.
2. custom _init_weight function will cause from_pretrained to load random weigh. The bug is fixed in https://github.com/huggingface/transformers/pull/43768.
3. MoE impl is refactored significantly to abandon the nn.modulelist approach for expert weight and changed to [E, I, H] 3-D tensor for better grouped gemm capability (but add complexity of model loading/saving which is still module list style for OSS models released before v5).

### Motivation

transformers v5 is a major update (See https://huggingface.co/blog/transformers-v5).

We want to support it to evolve with the oss model commnuity.

### Your contribution

TBA

**Comment by piyifan123** [comment]
some other comments from supporting transformers v5 in an internal inference engine. we should examine if these issues are applicable here.

```
 1. apply_chat_template now returns a dict by default
 The biggest pervasive change. In v5, apply_chat_template returns a dict instead of a string/list. You need to pass return_dict=False everywhere to preserve the old behavior. This affected ~15+ call sites across input processors, accuracy tests, quantization scripts, etc.
 2. config.rope_theta moved into config.rope_parameters
 RoPE configuration is now nested under a rope_parameters dict rather than being a top-level config attribute. The migration pattern used throughout:
 python
# Old
rope_theta = config.rope_theta

# New (backwards-compatible)
default_rope_theta = getattr(config, "rope_theta", 10000.0)
rope_params = getattr(config, "rope_parameters", None)
rope_theta = rope_params.get("rope_theta", default_rope_theta) if rope_params else default_rope_theta

 Same applies to partial_rotary_factor (seen in GLM4).
 3. VL models now use text_config sub-config
 Vision-language models (Qwen2VL, Qwen2.5VL, Qwen3VL, Pistis3.5) now nest the language model config under config.text_config instead of using the top-level config directly. The fix: text_config = getattr(config, "text_config", config) before passing to the language model constructor. Attributes like tie_word_embeddings also need to be propagated from parent config to text_config.
 4. AutoModelForVision2Seq removed
 Replaced by AutoModelForImageTextToText. Direct drop-in rename.
 5. T5Tokenizer / T5TokenizerFast removed
 These tokenizer classes were removed; references need to change to PreTrainedTokenizerFast.
 6. MoE weight layout change: gate_up_proj fused
 In v5, Qwen3 MoE experts combine gate_proj and up_proj into a single gate_up_proj tensor (similar to how many models fuse QKV). This requires updating weight mappings and passing is_gate_up_proj=True to the MoE block constructor.
 7. MoE forward return type change
 HF's SparseMoeBlock.forward now returns just the hidden states tensor instead of a (hidden_states, router_logits) tuple. The test changed from hf_output, _ = hf_moe(...) to hf_output = hf_moe(...).
 8. config.max_length now required
 Some model configs now expect a max_length attribute that wasn't previously needed. Tests had to add setattr(config, "max_length", 32768).
 9. BOS token handling change
 The old check hasattr(self.tokenizer, "add_bos_token") and self.tokenizer.add_bos_token no longer works. Replaced with explicit checking if the first token ID equals bos_token_id.
 10. sglang compatibility (Janus Pro)
 sglang's internal janus_pro config module breaks with transformers v5 due to AutoImageProcessor.register signature changes. The workaround is a dummy module stub injected into sys.modules before importing sglang.
 11. Dependency cascade
 The upgrade also pulled in newer versions of huggingface-hub (0.36→1.4.1), datasets (2.14→4.6), typer/typer-slim, and transitioned dependencies from requests to httpx. huggingface-hub also now depends on shellingham and typer-slim.
 12. Test infrastructure: version switching for vLLM compatibility
 Since vLLM doesn't yet support transformers v5, the accuracy test harness dynamically installs transformers 4.57.1 before running vLLM, then switches back. A vendored copy of modeling_qwen3_next.py from transformers 4.57.6 was added for unit tests that need the old HF reference implementation.
```
